### ISSUE TITLE
Review and update SOP for giving contributor keys and UUID

### ISSUE THREAD

**Issue by lauraclarke** [ebi-ait/hca-ebi-wrangler-central#78]
## Purpose
Review the current SOP for giving contributors access keys for the hca-util and UUID for the upload area so they can contribute data to the DCP. 

Current SOP: https://ebi-ait.github.io/hca-ebi-wrangler-central/SOPs/contributor_dataset_SOP.html#raw-data-upload-fastq

Propose changes as required to minimise the risk of exposure while bearing mind the ease and convenience of the current system to the contributor. If we choose to keep the current SOP, that may be ok, but I want that decision to be made having decided that the level of risk is acceptable.

## Acceptance Criteria
 - [ ] A brief written proposal outlining 3 possible solutions for how we transmit keys and UUID's to contributors.
- [ ] Pros and Cons of each of the possible solutions
- [ ] Favoured solution clearly highlighted
- [ ] Circulated to the team for comments and feedback
- [ ] final decision on new SOP
- [ ] SOP documentation updated to reflect any new processes

## Analysis TLDR
I've reviewed the security access of the keys that we distribute and believe they are sufficient. We could make the details of the **AWS key** public without any negative impact, but should protect the **folder names** to protect from malicious vandalism.

I believe our current SOP is sufficient, since these details are already communicated separately. Although we could make UX changes to the emails containing secure information so that they are not seen as the start of a conversations with the wranglers.

We should keep track of any folders granted additional permission over the default, especially download permission, and revoke these when they are no longer required.
(@MightyAx)

**Comment by mshadbolt** [comment]
@lauraclarke Is it necessary to go through the whole process above?

are we happy with the solution where we email the uuid separately and don't include wrangler-team?

**Comment by lauraclarke** [comment]
@clairerye what is your view?

**Comment by clairerye** [comment]
Could you briefly explain how that works - are the secret/keys still sent in the mail email or are there an additional 2 emails sent outside the main wrangler-team email correspondence chain?
I think it would be good to get some wider in put on whether this solution is reasonable in terms of trade off of risks versus ease, but happy for you not to present more than one option, or heavily detail it if you think this is the way we should go.

**Comment by MightyAx** [comment]
### TLDR
I've reviewed the security access of the keys that we distribute and believe they are sufficient. We could make the details of the **AWS key** public without any negative impact, but should protect the **folder names** to protect from malicious vandalism. I believe our current SOP is sufficient, although further steps could be made in future if they are required.

### Detail
The policy assigned to the **hca-contributor AWS key** is to only list items in the bucket not in the root folder. This means without knowing the folder name the **hca-contributor key** is completely safe to distribute publicly.

Once a **folder name** is know the **hca-contributor key** only has access to add or delete files, so even with the **folder name**, there is no risk of sensitive data being stolen or leaked. The largest possible impact of a malicious 3rd party gaining access to the **folder name** is deleing data, or overwriting it with false or corrupted data, essentially vandalising that copy.

We can mitigate the leaking of the **folder name** by educating the contributor to understand that the folder UUID is secret and should only be shared with people who have access to the data. Our current SOP already emphasise this, and so seems sufficient.

### Future Changes
It would be ideal if the **folder name**, and to lesser extent the **AWS key**, are not sent via email, since email exchanges should be considered as insecure, and leave our data operations open to vandalism. I believe this can wait until we have a credible threat of vandalism or because we wish to improve the user experience of contributor’s sending data.

**Comment by mshadbolt** [comment]
I think one suggestion was to not cc group emails when communicating the folder name and keys, or putting the folder name and keys in separate emails, do you think these things are necessary or we should just continue with one email with all the information and CC the wrangler-team as usual

**Comment by clairerye** [comment]
I like the idea of communicating the folder name and keys separately. If I understand the above analysis correctly, its actually the folder name that needs to be kept more private than the actual keys.

As I understand it, depending on the privileges that the folder was set up with, then it is possible to download all the data within the folder if someone got ahold of both the keys and folder name. But this is not the default and would only happen if whoever setup the folder gave it elevated privileges.

**Comment by MightyAx** [comment]
>  @clairerye: If I understand the above analysis correctly, its actually the folder name that needs to be kept more private than the actual keys.

Correct.

@mshadbolt I agree, separate emails seems like a good solution to mitigating the threat of the folder-name being buried in an email thread. We could mention the AWS key in the normal instructions and be specific with the contributor about the impact of the secret folder knowledge: "You will receive an separate email with your secret folder name, keep this confidential. In the unlikely event of a data breach, the folder name and AWS key would not give someone access to your data but **could be used to vandalise your submission.**"

We could probably also have the folder name email come from an account like do-not-reply@eci.ac.uk or similar, and use other UX triggers to signal that this email is not the start of a conversation with the wrangler-team. (Impersonal Language, "Do not reply", "this account is not monitored", etc)

> @clairerye depending on the privileges that the folder was set up with, then it is possible to download all the data within the folder

Good point, the the operations team can grant certain aws-keys additional permissions to folders, such as ability to download.
Ideally we can get AWS to report on folders that have more than the default permissions.
Otherwise we should start keeping track of which folders have been granted additional permissions, so we can set them back to default after the extra permissions are not needed.

I'll update these key points in ticket details above.

**Comment by lauraclarke** [comment]
https://github.com/ebi-ait/hca-ebi-wrangler-central/pull/173 for the SOP update
